### PR TITLE
Add Nix derivation for building

### PR DESCRIPTION
**Pull request by b123400**
Now you can build with a single command `nix-build`~~

### CODE REVIEW

**Comment by cheungpat** [comment]
Question (I have no background in nix):

Do we need to keep deps.nix and glide.lock in sync? if yes, perhaps we need to add a build target in `Makefile` so that we will update both files.

File: README.md [hunk 1]
```diff
@@ -69,6 +69,8 @@ If using Mac OS X, you can get the ZeroMQ dependencies using Homebrew:
 $ brew install libsodium zeromq czmq
 ```
 
+Or you can use Nix mentioned below, which also manages system dependencies.
```
Comment (cheungpat) [inline README.md]: I’d prefer to keep this section simple and don’t burden contributors with choices here.

Like I said, people who use nix should know what they are doing and they will just run the command below.

**Comment by b123400** [comment]
Updated, now it downloads deps using `glide` before build.

**Comment by cheungpat** [comment]
I tried building skygear-server with nix and it works flawlessly. That’s really simple and great.

I also added `result-bin` to `.gitignore`.

**Comment by b123400** [comment]
Thx! Forgot about gitignore...

**Pull request merged**